spec/integration/callbacks_spec.rb

```diff
@@ -447,4 +447,88 @@ class PreviouslyPersistedPerson
       expect(saved_person.previously_persisted_value).to be_truthy
     end
   end
+
+  context 'saved_change_to_attribute and attribute_before_last_save' do
+    class TestSCTAAndABLSInCallbacks
+      include Mongoid::Document
+
+      field :name, type: String
+      field :age, type: Integer
+
+      set_callback :save, :before do |doc|
+        [:name, :age].each do |attr|
+          saved_change_to_attribute_values_before[attr] += [saved_change_to_attribute(attr)]
+          attribute_before_last_save_values_before[attr] += [attribute_before_last_save(attr)]
+          will_save_change_to_attribute_values_before[attr] += [will_save_change_to_attribute?(attr)]
+        end
+      end
+
+      set_callback :save, :after do |doc|
+        [:name, :age].each do |attr|
+          saved_change_to_attribute_values_after[attr] += [saved_change_to_attribute(attr)]
+          attribute_before_last_save_values_after[attr] += [attribute_before_last_save(attr)]
+        end
+      end
+
+      def saved_change_to_attribute_values_before
+        @saved_change_to_attribute_values_before ||= Hash.new([])
```

Comment:
I think this copies the same array object into all new hash values which is generally not desirable. Unless this is specifically desired this should use the `new` with block syntax.

Reply:
Done